### PR TITLE
Rhamp 3215 unit tests

### PR DESCRIPTION
**Pull request by lholmquist**
@nialldonnellyfh so for some reason, my supertest is timing out.

### CODE REVIEW

File: test/unit/test-main.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+// mock out fh-mbaas-api to allow verification in tests
+var mockMbaasApi;
+var mockSys;
+var mockFhMiddleware;
+var mockCloud;
+var mockError;
+var mockSysHandler = function sys(){};
+var mockMbaasHandler = function mbaas(){};
+var mockFhMiddlewareHandler = function fhmiddleware(){};
```
Comment (nialldonnellyfh) [inline test/unit/test-main.js:26]: This is  an express middleware function so the signature needs to be function(req, res, next){} and it has to call next as a function.

File: test/unit/test-main.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+var mockCloudHandler = function cloud(){};
+var mockErrorHandler = function errorHandler(){};
+var createMockMbaasApi = function() {
+  return {
+    mbaasExpress: sinon.spy(function() {
```
Comment (nialldonnellyfh) [inline test/unit/test-main.js:31]: For the supertest, proxyquire needs to globally mock the fh-mbaas-api with the '@global': true, flag

https://github.com/thlorenz/proxyquire#globally-override-require

**Comment by nialldonnellyfh** [comment]
Instead of using the custom empty functions I would recommend using sinon.stub. I find it really useful for stubbing out methods.

**Comment by lholmquist** [comment]
@nialldonnellyfh added another commit, with a new test file just to make things a little cleaner.  hven't added the sinon stubs yet though,  i expected this to work,  but it doesn't seem to

**Comment by lholmquist** [comment]
closing to go against master

**Pull request closed without merging**